### PR TITLE
Search lambda: add "freeform" action

### PR DESCRIPTION
**Pull request by nl0**
# Description

Add "freeform" search action which passes the "body" param to elastic as is.

# TODO

- [x] Unit tests
- [x] [Changelog](CHANGELOG.md) entry

### CODE REVIEW

**Comment by nl0** [comment]
@sir-sigurd are these changes enough or am i missing smth?

File: lambdas/search/index.py [hunk 1]
```diff
@@ -229,6 +229,10 @@ def lambda_handler(request):
             'hits.total',
             'aggregations.objects.buckets.latest.hits.hits._source',
         )
+    elif action === 'freeform':
```
Comment (sir-sigurd) [inline lambdas/search/index.py:232]:
```suggestion
    elif action == 'freeform':
```

Comment (nl0) [reply lambdas/search/index.py]: pardon my js

**Comment by fiskus** [comment]
Can you provide `body` example, please?

`/search?index=_all&filter_path=hits&action=freeform&body={"query"%3A{"match"%3A{"message"%3A"instruments"}}}` has 0 hits

**Comment by nl0** [comment]
> Can you provide `body` example, please?
> 
> `/search?index=_all&filter_path=hits&action=freeform&body={"query"%3A{"match"%3A{"message"%3A"instruments"}}}` has 0 hits

try `{"query": { "match_all": {} } }`

**Review by sir-sigurd — APPROVED** [review]

File: lambdas/search/tests/test_search.py [hunk 2]
```diff
@@ -444,3 +444,41 @@ def _callback(request):
...
+            'from': 0,
+            'body': body,
+            'filter_path': 'hits',
+        }
+        url = 'https://www.example.com:443/idx1,idx2/_search?' + urlencode({
+            '_source': 'false',  # must match JSON; False will fail match_querystring
+            'size': 0,
+            'from': 0,
+            'filter_path': 'hits',
+            'terminate_after': 10000,
```
Comment (sir-sigurd) [inline lambdas/search/tests/test_search.py:466]: Might make sense either mock it or use a variable where it's defined.

**Review by sir-sigurd — APPROVED** [review]

File: lambdas/search/tests/test_search.py [hunk 3]
```diff
@@ -444,3 +444,41 @@ def _callback(request):
...
+            'size': 0,
+            'from': 0,
+            'body': body,
+            'filter_path': 'hits',
+        }
+        url = 'https://www.example.com:443/idx1,idx2/_search?' + urlencode({
+            '_source': 'false',  # must match JSON; False will fail match_querystring
+            'size': 0,
+            'from': 0,
+            'filter_path': 'hits',
```
[the remark this answers is not in the document]
Comment (nl0) [reply lambdas/search/tests/test_search.py]: i've just reused the relevant parts from `query`

**Pull request merged**